addons/l10n_gcc_invoice/views/report_invoice.xml

```diff
@@ -344,27 +344,40 @@
                                     </td>
                                 </tr>
                                 <t t-set="tax_totals" t-value="o.tax_totals"/>
-                                <t t-foreach="tax_totals['subtotals']" t-as="subtotal">
-                                    <t t-set="subtotal_to_show" t-value="subtotal['name']"/>
+                                <t t-foreach="enumerate(tax_totals['subtotals'])" t-as="subtotal">
+                                    <t t-set="subtotal_to_show" t-value="subtotal[1]['name']"/>
                                     <!-- copy-pasted template "account.tax_groups_totals" with reversed columns order -->
-                                    <t t-foreach="tax_totals['groups_by_subtotal'][subtotal_to_show]" t-as="amount_by_group">
+                                    <t t-foreach="enumerate(tax_totals['groups_by_subtotal'][subtotal_to_show])" t-as="amount_by_group">
                                         <tr>
+                                            <t t-set="group_name_arab" t-value="o_sec.tax_totals['groups_by_subtotal'][o_sec.tax_totals['subtotals'][subtotal[0]]['name']][amount_by_group[0]]['tax_group_name']"/>
```

Comment:
It seems you define the variable group_name_arab for only displaying it once, is it therefore necessary ? Plus it's defined quite far in the code compared to where it's used